Source/EmbeddedBoundary/ParticleScraper.H

```diff
@@ -172,10 +172,14 @@ scrapeParticles (PC& pc, const amrex::Vector<const amrex::MultiFab*>& distance_t
 
                 // the closest point on the surface to pos is pos - grad phi(pos) * phi(pos)
                 amrex::RealVect pos;
+#if (defined WARPX_DIM_3D)
                 pos[0] = xp - normal[0]*phi_value;
                 pos[1] = yp - normal[1]*phi_value;
                 pos[2] = zp - normal[2]*phi_value;
-
+#elif (defined WARPX_DIM_XZ)
+                pos[0] = xp - normal[0]*phi_value;
+                pos[1] = yp - normal[1]*phi_value;
+#endif
                 DistanceToEB::normalize(normal);
 
                 if (phi_value < 0.0) {
```

Comment:
While we're at it, should we move all the calculation inside this `if`?

Reply:
Good idea :)